src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs

```diff
@@ -370,6 +370,11 @@ internal void SetPossibleRedirectForLocationHeader(string location)
                     {
                         SetCookieOption(newUri);
                     }
+
+                    if (newUri.Scheme == Uri.UriSchemeHttp && _requestMessage.RequestUri.Scheme == Uri.UriSchemeHttps)
+                    {
+                        EventSourceTrace($"Insecure https to http redirect from {_requestMessage.RequestUri} to {newUri} blocked.");
```

Comment:
This is unfortunately a relatively expensive operation.  This is translated by the compiler into:
```C#
EventSourceTrace(string.Format("Insecure https to http redirect from {0} to {1} blocked.", new object[] { _requestMessage.RequestUri, newUri }));
```
so even if tracing is disabled, we still end up allocating that object[], and we still end up doing the string format.

It should instead do something like:
https://github.com/rmkerr/corefx/blob/c1a2eddf4ee509b1e591f8e63ad7429625e42b20/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs#L190
where the argument is passed in as an argument, and all of the work is deferred until in the EventSourceTrace method it checks for whether tracing is enabled.  I don't recall for sure, but I don't think there's currently an EventSourceTrace overload that takes two arguments to be substituted; assuming not, you could just use a value tuple, e.g.
```C#
EventSourceTrace("Insecure https to http redirect: {0}", (_requestMessage.RequestUri, newUri));
```

Reply:
I'm used to thinking of that kind of string operation as "free", but obviously that isn't the case with format strings. I'll make the change, and the cost of format strings is something I'll keep in mind!